drivers/sdhc/sdhc_renesas_ra.c

```diff
@@ -0,0 +1,736 @@
+			/*  0 = SDSC, 1 = SDHC or SDXC */
+			priv->sdmmc_ctrl.sector_addressing =
+				(response.r3.card_capacity_status > 0U);
+			priv->sdmmc_ctrl.device.card_type = SDMMC_CARD_TYPE_SD;
+		}
+		priv->sdmmc_ctrl.initialized = true;
+		break;
+	case SD_SWITCH:
+		/* Check app cmd */
+		if (priv->app_cmd && cmd->opcode == SD_APP_SET_BUS_WIDTH) {
+			/* ACMD41*/
+			ra_cmd.opcode |= SDHI_PRV_CMD_C_ACMD;
+			ret = sdhc_ra_send_cmd(priv, &ra_cmd, retries);
+			if (ret < 0) {
+				goto end;
+			}
+		} else {
+			/* SD SWITCH CMD6*/
+			fsp_err = r_sdhi_read_and_block(&priv->sdmmc_ctrl, ra_cmd.opcode,
+							ra_cmd.arg, data->block_size);
+			ret = err_fsp2zep(fsp_err);
+			if (ret < 0) {
+				goto end;
+			}
+			memcpy(ra_cmd.data, priv->sdmmc_ctrl.aligned_buff, 8);
+			sdmmc_event[priv->channel].transfer_completed = false;
+			break;
+		}
+		break;
+
+	/* Read write with data */
+	case SD_APP_SEND_SCR:
+		ra_cmd.opcode = cmd->opcode | SDHI_PRV_CMD_C_ACMD;
+		fsp_err = r_sdhi_read_and_block(&priv->sdmmc_ctrl, ra_cmd.opcode, ra_cmd.arg,
+						data->block_size);
+
+		if (fsp_err != 0) {
+			ret = -ETIMEDOUT;
+			goto end;
+		}
+		memcpy(ra_cmd.data, priv->sdmmc_ctrl.aligned_buff, 8);
+		sdmmc_event[priv->channel].transfer_completed = false;
+		break;
+	case SD_READ_SINGLE_BLOCK:
+	case SD_READ_MULTIPLE_BLOCK:
+		/* Configure the transfer interface for reading.*/
+		fsp_err = r_sdhi_transfer_read(&priv->sdmmc_ctrl, ra_cmd.sector_count,
+					       (uint32_t)priv->fsp_config.block_size, ra_cmd.data);
+		ret = err_fsp2zep(fsp_err);
+		if (ret < 0) {
+			goto end;
+		}
+
+		r_sdhi_read_write_common(&priv->sdmmc_ctrl, ra_cmd.sector_count,
+					 (uint32_t)priv->fsp_config.block_size, ra_cmd.opcode,
+					 ra_cmd.arg);
+
+		/* Verify card is back in transfer state after write */
+		ret = k_sem_take(&sdmmc_event[priv->channel].transfer_sem,
+				 K_MSEC(ra_cmd.timeout_ms));
+		if (ret < 0) {
+			LOG_ERR("Can not take sem!");
+			goto end;
+		}
+
+		if (!sdmmc_event[priv->channel].transfer_completed) {
+			ret = -EIO;
+			goto end;
+		}
+
+		sdmmc_event[priv->channel].transfer_completed = false;
+		break;
+
+	case SD_WRITE_SINGLE_BLOCK:
+	case SD_WRITE_MULTIPLE_BLOCK:
+
+		fsp_err = r_sdhi_transfer_write(&priv->sdmmc_ctrl, ra_cmd.sector_count,
+						(uint32_t)priv->fsp_config.block_size, ra_cmd.data);
+		ret = err_fsp2zep(fsp_err);
+		if (ret < 0) {
+			goto end;
+		}
+		/* Send command with data for reading */
+		r_sdhi_read_write_common(&priv->sdmmc_ctrl, ra_cmd.sector_count,
+					 (uint32_t)priv->fsp_config.block_size, ra_cmd.opcode,
+					 ra_cmd.arg);
+
+		/* Verify card is back in transfer state after write */
+		ret = k_sem_take(&sdmmc_event[priv->channel].transfer_sem,
+				 K_MSEC(ra_cmd.timeout_ms));
+		if (ret < 0) {
+			goto end;
+		}
+
+		if (!sdmmc_event[priv->channel].transfer_completed) {
+			ret = -EIO;
+			goto end;
+		}
+
+		sdmmc_event[priv->channel].transfer_completed = false;
+		break;
+
+	default:
+		ret = -ENOTSUP;
```

Comment:
Is this intentional? IE are other SD opcodes not supported by this driver?

Reply:
You're correct, other SD opcodes not supported by this driver. I added a comment here for more clearly